File: python/Ejercicio_9.py
# Diccionario para almacenar las facturas
facturas = {}
cantidad_cobrada = 0
cantidad_pendiente = 0

while True:
    print("\nSeleccione una opción:")
    print("1. Añadir nueva factura")
    print("2. Pagar una factura")
    print("3. Terminar")
    opcion = input("Ingrese su opción: ")

    if opcion == "1":
        # Añadir una nueva factura
        numero_factura = input("Ingrese el número de la factura: ")
        coste = float(input("Ingrese el coste de la factura: "))
        facturas[numero_factura] = coste
        cantidad_pendiente += coste
        print(f"Factura {numero_factura} añadida con un coste de {coste:.2f}.")

    elif opcion == "2":
        # Pagar una factura existente
        numero_factura = input("Ingrese el número de la factura a pagar: ")
        if numero_factura in facturas:
            coste = facturas.pop(numero_factura)
            cantidad_cobrada += coste
            cantidad_pendiente -= coste
            print(f"Factura {numero_factura} pagada con un coste de {coste:.2f}.")
        else:
            print(f"La factura {numero_factura} no existe.")

    elif opcion == "3":
        # Terminar el programa
        print("Terminando el programa.")
        break

    else:
        print("Opción no válida. Intente nuevamente.")

    # Mostrar las cantidades cobradas y pendientes
    print(f"Cantidad cobrada hasta el momento: {cantidad_cobrada:.2f}")
    print(f"Cantidad pendiente de cobro: {cantidad_pendiente:.2f}")

File: java/Ejercicio_9.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Ejercicio_9 {
    public static void main(String[] args) {
        Map<String, Double> facturas = new HashMap<>();
        double cantidadCobrada = 0;
        double cantidadPendiente = 0;

        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("\nSeleccione una opción:");
            System.out.println("1. Añadir nueva factura");
            System.out.println("2. Pagar una factura");
            System.out.println("3. Terminar");
            System.out.print("Ingrese su opción: ");
            String opcion = scanner.nextLine();

            if (opcion.equals("1")) {
                // Añadir una nueva factura
                System.out.print("Ingrese el número de la factura: ");
                String numeroFactura = scanner.nextLine();
                System.out.print("Ingrese el coste de la factura: ");
                double coste = Double.parseDouble(scanner.nextLine());
                facturas.put(numeroFactura, coste);
                cantidadPendiente += coste;
                System.out.printf("Factura %s añadida con un coste de %.2f.%n", numeroFactura, coste);
            } else if (opcion.equals("2")) {
                // Pagar una factura existente
                System.out.print("Ingrese el número de la factura a pagar: ");
                String numeroFactura = scanner.nextLine();
                if (facturas.containsKey(numeroFactura)) {
                    double coste = facturas.remove(numeroFactura);
                    cantidadCobrada += coste;
                    cantidadPendiente -= coste;
                    System.out.printf("Factura %s pagada con un coste de %.2f.%n", numeroFactura, coste);
                } else {
                    System.out.printf("La factura %s no existe.%n", numeroFactura);
                }
            } else if (opcion.equals("3")) {
                // Terminar el programa
                System.out.println("Terminando el programa.");
                break;
            } else {
                System.out.println("Opción no válida. Intente nuevamente.");
            }

            // Mostrar las cantidades cobradas y pendientes
            System.out.printf("Cantidad cobrada hasta el momento: %.2f%n", cantidadCobrada);
            System.out.printf("Cantidad pendiente de cobro: %.2f%n", cantidadPendiente);
        }

        scanner.close();
    }
}
